biorxiv/pca_association_experiment/biorxiv_pca_plots.py

```diff
@@ -0,0 +1,810 @@
+
+from pca_plot_helper import *
+
+
+# In[2]:
+
+
+journal_map_df = pd.read_csv("../exploratory_data_analysis/output/biorxiv_article_metadata.tsv", sep="\t")
+journal_map_df.head()
+
+
+# # PCA the Documents
+
+# Run PCA over the documents. Generates 50 principal components, but can generate more or less.
+
+# In[3]:
+
+
+n_components = 50
+random_state = 100
+
+
+# In[4]:
+
+
+biorxiv_articles_df = pd.read_csv(
+    "../word_vector_experiment/output/word2vec_output/biorxiv_all_articles_300.tsv.xz", 
+    sep="\t"
+)
+
+
+# In[5]:
+
+
+reducer = PCA(
+    n_components = n_components,
+    random_state = random_state
+)
+
+embedding = reducer.fit_transform(
+    biorxiv_articles_df[[f"feat_{idx}" for idx in range(300)]].values
+)
+
+pca_df = (
+    pd.DataFrame(embedding, columns=[f"pca{dim}" for dim in range(1, n_components+1, 1)])
+    .assign(document=biorxiv_articles_df.document.values.tolist())
+    .merge(journal_map_df[["category", "document", "doi"]], on="document")
+)
+
+latest_version = (
+    pca_df
+    .groupby("doi", as_index=False)
+    .agg({"document":"last"})
+    .document
+    .values
+    .tolist()
+)
+
+pca_df = (
+    pca_df
+    .query(f"document in {latest_version}")
+    .reset_index(drop=True)
+)
+
+pca_df.head()
+
+
+# In[6]:
+
+
+reducer.explained_variance_
+
+
+# In[7]:
+
+
+reducer.explained_variance_ratio_
+
+
+# In[8]:
+
+
+(
+    pca_df
+    .category
+    .sort_values()
+    .unique()
+)
+
+
+# # Plot the PCA axes
+
+# This section visualizes PCA axes and attempts to provide an explanation for each plotted PC.
+# Give there are 50 pcs generated this notebook/section will only focus on PCs that generate a clear enough signal.
+
+# ## Basis PCs
+
+# When analyzing PCs, it is generally a good idea to start off with the PCs that are easily distinguishable and then branch out to PCs that are harder to interpret. This section focuses on PCs: 1,2,3,4 and 5.
+# 
+# These PCs correspond to the following concepts:
+# 
+# | PC | Concept (negative vs positive)|
+# | --- | --- |
+# | PCA1 | Quantitative Biology vs Molecular Biology |
+# | PCA2 | Neuroscience vs Genomics |
+# | PCA3 | Sequencing vs Disease |
+# | PCA4 | Microbiology vs Cell Biology |
+# | PCA5 | RNA-seq vs Evolutional Biology | 
+
+# In[9]:
+
+
+global_color_palette = [
+    '#a6cee3','#1f78b4',
+    '#b2df8a','#33a02c',
+    '#fb9a99'
+]
+
+
+# ### PCA1 vs PCA2
+
+# In[10]:
+
+
+display_clouds(
+    'output/word_pca_similarity/figures/pca_01_cossim_word_cloud.png',
+    'output/word_pca_similarity/figures/pca_02_cossim_word_cloud.png'
+)
+
+
+# These word clouds depict the following concepts: quantitative biology vs molecular biology (left) and genomics vs neuroscience (right). The cells below provide evidence for the previous claim
+
+# In[11]:
+
+
+selected_categories = [
+    'biochemistry', 'bioinformatics',
+    'cell biology', 'neuroscience',
+    'scientific communication'
+]
+
+
+# In[12]:
+
+
+generate_scatter_plots(
```

Comment:
Interesting..looks like PC2 is more distinguishing for neuroscience compared to PC1